exp/views/study.py

```diff
@@ -200,6 +202,64 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class StudyParticipantEmailView(ExperimenterLoginRequiredMixin, PermissionRequiredMixin, generic.DetailView):
+    '''
+    StudyParticipantEmailView allows user to send a custom email to a participant.
+    '''
+    model = Study
+    permission_required = 'studies.can_edit_study'
+    raise_exception = True
+    template_name = 'studies/study_participant_email.html'
+
+    def get_context_data(self, **kwargs):
+        """
+        Adds email to the context_data dictionary
+        """
+        context = super().get_context_data(**kwargs)
+        context['sender'] = settings.EMAIL_FROM_ADDRESS
+        context['participants'] = self.get_study_participants()
+        return context
+
+    def get_study_participants(self):
+        '''
+        Restricts list to participants that have responded to this study as well as participants
+        that have given their permission to be emailed personally
+        '''
+        return User.objects.filter(Q(children__response__study__id=self.get_object().id) & Q(email_personally=True)).distinct()
```

Comment:
You can just do `Q(children__response__study=self.get_object())`. The object is already queried and built when you call get_object so doing the id doesn't really save anything.